pkg/virt-launcher/virtwrap/agent-poller/agent_parser.go

```diff
@@ -249,11 +250,13 @@ func MergeAgentStatusesWithDomainData(domInterfaces []api.Interface, interfaceSt
 	}
 
 	aliasesCoveredByAgent := []string{}
-	// Add alias from domain to interfaceStatus
 	for i, interfaceStatus := range interfaceStatuses {
 		if alias, exists := aliasByMac[interfaceStatus.Mac]; exists {
 			interfaceStatuses[i].Name = alias
+			interfaceStatuses[i].InfoSource = netvmispec.InfoSourceDomainAndGA
```

Comment:
In order to move the whole logic to `netstat.go`
We need first this fix https://github.com/kubevirt/kubevirt/pull/6926

I tried to start with it, and saw that it uncover the leakage in case the mac is changed
https://github.com/kubevirt/kubevirt/pull/6744 (last comment shows it)